### PR TITLE
MES-6277 Waiting Room page

### PR DESCRIPTION
**Pull request by Cullima**
## Description
[MES-6277](https://jira.dvsacloud.uk/browse/MES-6277)
- Port Waiting Room page and components and make common for all categories

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image

### CODE REVIEW

File: src/app/pages/waiting-room/__tests__/waiting-room.page.spec.ts [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+                communicationMethod: 'Email',
+                conductedLanguage: 'Cymraeg',
+              },
+            },
+          },
+        })),
+      ],
+      providers: [
+        { provide: Router, useValue: routerSpy },
+        { provide: NavParams, useFactory: () => NavParamsMock.instance() },
```
Comment (andrewsetterfield) [inline src/app/pages/waiting-room/__tests__/waiting-room.page.spec.ts]: are you really using nav params?

**Comment by andrewsetterfield** [comment]
can you change some text to be consistant.  'WAITING_ROOM_PAGE' to 'WaitingRoomPage' on lines121 and 144 of page-names-constant

File: src/app/pages/waiting-room/__tests__/waiting-room.page.spec.ts (hunk 1 above)
Comment (Cullima) [reply src/app/pages/waiting-room/__tests__/waiting-room.page.spec.ts]: removing

**Review by andrewsetterfield — APPROVED** [review]

File: src/app/pages/waiting-room/waiting-room.page.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+  formGroup: FormGroup;
+
+  subscription: Subscription;
+
+  merged$: Observable<boolean | string | JournalData>;
+
+  constructor(
+    store$: Store<StoreModel>,
+    public router: Router,
+    public platform: Platform,
```
Comment (matthew2564) [inline src/app/pages/waiting-room/waiting-room.page.ts]: platform and authenticationProvider, if only needed for basepage could remove access modifiers

**Review by matthew2564 — APPROVED** [review]

File: src/app/pages/waiting-room/waiting-room.page.ts (hunk 2 above)
Comment (Cullima) [reply src/app/pages/waiting-room/waiting-room.page.ts]: fair comment....will remove

**Pull request merged**